File: qna/src/main/java/com/example/rag/qna/adapter/output/opinion/OpinionEntity.java
package com.example.rag.qna.adapter.output.opinion;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "opinions")
@NoArgsConstructor
@Getter
@Setter
public class OpinionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "question_id", nullable = false)
    private String questionId;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "content", nullable = false, length = 1000)
    private String content;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Version
    private int version; // 낙관적 락

    public OpinionEntity(String questionId, String userId, String title, String content, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.questionId = questionId;
        this.userId = userId;
        this.title = title;
        this.content = content;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}

File: qna-persistence/src/main/kotlin/com/example/rag/qna/adapter/output/opinion/OpinionEntity.kt
package com.example.rag.qna.adapter.output.opinion

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "opinions")
class OpinionEntity(
    @Column(name = "question_id", nullable = false)
    var questionId: String,

    @Column(name = "title", nullable = false)
    var title: String,

    @Column(name = "content", nullable = false, length = 1000)
    var content: String,

    @Column(name = "user_id", nullable = false)
    var userId: String,

    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime,

    @Column(name = "updated_at")
    var updatedAt: LocalDateTime = createdAt
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Version
    var version = 0 // 낙관적 락
}